Add the tutorial 'Ruby On Rails in Production'

vincent-the opened the pull request:
This PR adds the tutorial 'Ruby On Rails in Production'

I have read and understood the Contributor's Certificate of Origin at the end of the template and I hereby certify that I meet the contribution criteria described in it.

Signed-off-by: Vincent Thelang [info@vincent-thelang.de](mailto:info@vincent-thelang.de)

raphi-0901 commented on community-tutorials/ruby-on-rails-in-production/01-en.md:
```diff
@@ -0,0 +1,255 @@
+
+The important thing is that we change `USERNAME` in the name of our user. to show what it is, we run `whoami`.
+
+Once we have saved the file, we execute the following command to start NGINX.
```
Is this command missing?

vincent-the replied:
fixed in cb797f6

raphi-0901 commented on community-tutorials/ruby-on-rails-in-production/01-en.md:
````diff
@@ -0,0 +1,256 @@
+If you want to start from scratch, you can create a new rails app here with the name `rails_app` and execute the following command:
+
+```sh
+rails new rails_app
````
I get this error when I try this command:
```
root@v220211184008167421:~# rails new rails_app
Command 'rails' not found, but can be installed with:
apt install ruby-railties
```

Please add this to your tutorial.

vincent-the replied:
fixed in cb797f6

vincent-the commented on community-tutorials/ruby-on-rails-in-production/01-en.md:
````diff
@@ -0,0 +1,256 @@
+To finally compile the javascript files execute the following command:
+
+```sh
+RAILS_ENV=production rails assets:precompile
````
fixed in cb797f6

raphi-0901 commented:
I don´t know why, but I am able to run the compiler but I can´t see it in the browser. I´m getting no errors. I had access to it before we removed the default configuration of nginx.

vincent-the commented:
@raphi-0901 
> I don´t know why, but I am able to run the compiler but I can´t see it in the browser. I´m getting no errors. I had access to it before we removed the default configuration of nginx.

I have tested it again on the latest version, there the same behavior. I have remove the part with the deletion of the default config (
7bf2899). I think it is because of the settings of the default_server names.

raphi-0901 commented on community-tutorials/ruby-on-rails-in-production/01-en.md:
````diff
@@ -0,0 +1,269 @@
+rails --version
+```
+
+If it says something like `Rails 7.0.2`, you are ready to continue.
````
I think this is no problem, but I´m getting `Rails 6.1.3.1`

vincent-the replied:
Is not relevant which version, since this changes quickly I have removed the part.

raphi-0901 commented on community-tutorials/ruby-on-rails-in-production/01-en.md:
````diff
@@ -0,0 +1,269 @@
++ passenger_ruby /home/USERNAME/.rbenv/shims/ruby;
+```
+
+The important thing is that we change `USERNAME` in the name of our user. To show what it is, we run `whoami`.
````
Do I have to create an extra user for this with a home directory?

vincent-the replied:
Yes, i have added the part.

raphi-0901 commented:
I now tried it again from scratch and it still does not work. Now when I try reaching my server I just get the default nginx site.

vincent-the commented:
@raphi-0901 
> I now tried it again from scratch and it still does not work. Now when I try reaching my server I just get the default nginx site.

Hmm, i tried it with my domain and there it works as described above. I have changed it in the tutorial to a domain. 
I have added the part

raphi-0901 commented:
Are you using an extra domain or the default one provided by netcup?

vincent-the commented:
@raphi-0901 

In this case an external one but it should work the same way with the internal one from netcup.

raphi-0901 commented on community-tutorials/ruby-on-rails-in-production/01-en.md:
````diff
@@ -0,0 +1,269 @@
+
+```diff
+- passenger_ruby /usr/bin/passenger_free_ruby;
++ passenger_ruby /home/rails_user/.rbenv/shims/ruby;
````
I checked now and found out that the folder `shims` is not existing in my home directory. Are we perhaps forgetting something here?

vincent-the replied:
If I install rbenv under the user as described this works. You can try `rbenv rehash` once and see if it works now.

raphi-0901 commented on community-tutorials/ruby-on-rails-in-production/01-en.md:
````diff
@@ -0,0 +1,269 @@
+To finally compile the javascript files execute the following command:
+
+```sh
+RAILS_ENV=production rails assets:precompile
````
I can´t run this code with rails_user.

vincent-the replied:
I think you have not installed rbenv as rails_user. That is why these errors occur.

raphi-0901 commented on community-tutorials/ruby-on-rails-in-production/01-en.md:
````diff
@@ -0,0 +1,269 @@
+If you want to start from scratch, you can create a new rails app here with the name `rails_app` and execute the following command:
+
+```sh
+rails new rails_app
````
I can´t run this code with rails_user.

vincent-the replied:
I think you have not installed rbenv as rails_user. That is why these errors occur.

raphi-0901 replied:
I reran everything again and I had to change the permissions on `/var/www` to run this command without errors. 
Now, when I visit my server, I get the default nginx page again.

vincent-the replied:
Hmm strange, for me it works with the domain. Can you delete the default config again. Maybe you can also post your config. I have information from the following link. There it is also not documented differently.

https://www.phusionpassenger.com/library/deploy/nginx/deploy/ruby/

raphi-0901 replied:
I have now deleted the nginx defailt config file and now I´m getting the Rails error page. (I guess this is a success)

vincent-the replied:
Yes perfect. The error is displayed I think because the request is not routed to a controller.

raphi-0901 commented:
Hey!
Could you please add `Steps` to your headings  as shown in the [template file](https://github.com/netcup-community/community-tutorials/blob/main/TEMPLATE.md)?

vincent-the commented:
@raphi-0901 

> Hey! Could you please add `Steps` to your headings as shown in the [template file](https://github.com/netcup-community/community-tutorials/blob/main/TEMPLATE.md)?

Done ✅

netcupClaudiaM commented:
Hi vincent-the, thank you for contributing! It's been a while since you opened this PR and it probably is an open secret we're currently overwhelmed with reviewing all the awesome tutorials we got. Please be patient - we're about to publish tutorials this and next week. Thank you!
